api/authenticator.py

```diff
@@ -1804,6 +1822,13 @@ def __init__(self, library, integration, analytics=None):
             or self.DEFAULT_PASSWORD_LABEL
         )
 
+        self.oauth_enabled = self.HTTP_BASIC_OAUTH_ENABLED_DEFAULT
+        oauth_enabled = list(
+            filter(lambda setting: setting.key == self.HTTP_BASIC_OAUTH_ENABLED, library.settings)) \
+            or self.HTTP_BASIC_OAUTH_ENABLED_DEFAULT
+        if oauth_enabled:
+            self.oauth_enabled = oauth_enabled[0].bool_value or self.HTTP_BASIC_OAUTH_ENABLED_DEFAULT
```

Comment:
Is there a better way to get this setting? I tried calling it like the other settings here, `integration.setting(self.HTTP_BASIC_OAUTH_ENABLED).value` and `.bool_value` but nothing is returned.

Reply:
There are a bunch of class methods on `ConfigurationSetting` like `ConfigurationSetting.for_library(key, library)`, but they all flow through `ConfigurationSetting.for_library_and_externalintegration()`, which will create the setting if it doesn't already exist. I'm assuming that you just want to access the setting if it's there, not create it, so probably you do need to iterate / filter the settings on the library object you've got. That said, it may be more efficient to run the filter on the db side, so you may want to switch to using a chained `.filter()` SQLAlchemy call instead of a native Python iterable filter.